### ISSUE TITLE
Bug: OperationSelectorRoute with incr/decr ErrorRoute Broken

### ISSUE THREAD

**Issue by amcrn** [facebook/mcrouter#182]
**Summary**

If `OperationSelectorRoute` is used, and `incr`/`decr` are set to use `ErrorRoute`, the response from an `incr`/`decr` command is not the `ErrorRoute` value as expected, but instead it's `NOT_FOUND`.

Reproducible on mcrouter 0.35.0 (latest).

**How to Reproduce**

```
{
  "pools": {
    "main": {
      "servers": [
        "memcached1:11211",
        "memcached2:11211",
        "memcached3:11211"
      ]
    }
  },
  "route": {
    "type": "OperationSelectorRoute",
    "operation_policies": {
      "add": "ErrorRoute|MyErrorValueHere",
      "get": "ErrorRoute|MyErrorValueHere",
      "incr": "ErrorRoute|MyErrorValueHere",
      "decr": "ErrorRoute|MyErrorValueHere",
      "set": "ErrorRoute|MyErrorValueHere",
      "replace": "ErrorRoute|MyErrorValueHere",
      "append": "ErrorRoute|MyErrorValueHere",
      "prepend": "ErrorRoute|MyErrorValueHere",
      "touch": "ErrorRoute|MyErrorValueHere",
      "cas": "ErrorRoute|MyErrorValueHere",
      "delete": "ErrorRoute|MyErrorValueHere"
    }
  }
}
```

Note how `add`, `set`, etc. all respond as expected:

```
$ telnet localhost 11211
Trying ::1...
Connected to localhost.
Escape character is '^]'.

set foo 0 0 1
0
SERVER_ERROR MyErrorValueHere

add foo 0 0 1
0
SERVER_ERROR MyErrorValueHere
```

But `incr` and `decr` do not:

```
incr foo 1 
NOT_FOUND

decr foo 1
NOT_FOUND
```

**How to Reproduce**

`incr` and `decr` respond with `SERVER_ERROR MyErrorValueHere`.

**Comment by amcrn** [comment]
Hey @jmswen @andreazevedo (sorry for the direct mentions, but i saw your names in the recent commits list, so I thought I'd give ya'll a try):

Was wondering if anyone has had a chance to look at the GitHub issues I've filed in early March and onward? Please let me know if the information isn't up to snuff, or if I should be sending such inquiries/thoughts to a mailing list or IRC or Slack. Cheers :)

**Comment by andreazevedo** [comment]
Hello @amcrn,
You are right, we never return errors for `incr` and `decr`. 
Is it something that you need? If so, I can add an option to tweak this behavior.

**Comment by amcrn** [comment]
Yes, that'd be great! It's probably worth mentioning another strange behavior w/ incr+decr since it might impact the solution: See #186 `incr/decr an existing non-numeric value`.

I've also filed #180, #183, #184, #185, #186, #189 when someone has the chance to look them over. A mixture of doc suggestions, questions, feature requests, and bug reports.

Thanks a bunch!
